internal/cmd/integration-tests/configs/prom-gen/main.go

```diff
@@ -44,7 +44,11 @@ func main() {
 	}()
 	log.Printf("HTTP server on %s", listenAddress)
 
-	go func() { log.Fatal(server.ListenAndServe()) }()
+	go func() {
```

Comment:
I think you made changes in the wrong place. This `main.go is to run integration tests.

Robert already pointed at possibility of logging out the error before return it here. Could use something like `level.Error(s.log).Log("failed to start HTTP service", "msg", err.Error()`

https://github.com/grafana/alloy/blob/c6f4f3054b9f5032ee8d38b1570bf451dfd58fd3/internal/service/http/http.go#L154